text/0012-frameworks.md

```diff
@@ -0,0 +1,52 @@
+# Frameworks at Canopy
+- RFC PR: https://github.com/CanopyTax/frontend-rfc/pull/12
+
+## Definition
+There are a lot of front-end frameworks out there and we use several at Canopy. By design we've sought to avoid locking ourselves
+into a specific framework and becoming an `${frameworkX} shop`. This has come with a lot of pros and cons. Our goal is for the 
+best tools to "bubble up", rather than be selected and pushed "top down".
+
+## Why can we be so flexible with frameworks
+- [single-spa](https://github.com/CanopyTax/single-spa)
+- [SystemJS](https://github.com/systemjs/systemjs)
+- [SOFE](https://github.com/CanopyTax/sofe)
+
+### pros
+- Easy to build, experiment, and change our tech stack.
+- We aren't locked into Angular or React.
+- We can, theorectially, choose the best framework for the job.
+
+### cons
+- It could become the wild west of frameworks, where each squad uses frameworks/tools that no one else has any familiarity with.
+- Bundle size if multiple frameworks need to be downloaded
+- Could affect hiring (Squad X has an opening but they use Framework Y so we have to hire a Y specialist)
+
+## Squad vs Chapter vs Frontend-wide
+All chapters should standardize on our supported frameworks.
+
+When a new customer facing feature is being built it's important to recognize that introducing a new framework comes with a 
+lot of costs and we have to manage those costs across the entire front-end. New frameworks should be added with care as a 
+result it isn't a squad level decision.
+
+Introducing a new framework requires your colleagues to understand the framework paradigm in order to make changes, fix bugs, or properly 
+respond to an alert.
+
+
+## Supported Frameworks
+- React
+- AngularJS (legacy)
+
+We're currently moving away from AngularJS and have bubble-up standardized on React. We have a few experiments in Vue and 
+Svelte but they aren't something we're fully committed to introducing into our application yet.
```

Comment:
The section is an area I've been thinking about.

Do we prefer listing specific things we're doing, at the cost of then needing to come update the RFC when those specific things change? 

Or should we be more general and not worry about the specifics? E.g. in this case, should we just omit this section completely? And when a new employee is onboarded they can learn what framework(s) we've standardized on at that time?

Reply:
I debated the exact same thing. I felt that the cost of changing this area is pretty small and I wanted to open it up for discussion. 

I feel that we should call out the status of AngularJS, but it felt weird to do that without also calling out what we've standardized on.

Reply:
This isn't an exhaustive list of frameworks in our codebase. I like how it tells the story of howe we are trying to move away from AngularJs and who knows, in the future, perhaps React.

Reply:
This is meant to be an exhaustive list of fully supported frameworks. 

What I'm trying to get at is that we intentionally limit the amount of frameworks in production because of all the cons listed earlier. There needs to be a process where a framework can be implemented as an experiment (again limited) and eventually adopted as a free for all.

Example:
We have Svelte in our app as an experiment. If it makes sense to use Svelte elsewhere (and it's approved) and again then it would become a fully supported framework. Where everyone needs to know how to use it. Right now we only have React and AngularJS as fully supported. 

I don't know what the process of going from Experiment to fully supported looks like but it's something we collectively need to define. We don't want to make it too hard or too easy because of all the pros and cons mentioned above.

Reply:
Re-reading my message. I think what I should have said is the following:

I originally envisioned this to be an exhaustive list of frameworks that any squad could choose to use without needing any approval. Right now that list is just one: React, but I don't think it will always be this way. We need to figure out a process to move a framework from idea -> experiment -> full support in the next section.